OctopusDeploy/Calamari
# Use environment variable as path for package retention journal

This PR updates the location of the journal used for package retention from a hard-coded one to an environment variable `CalamariPackageRetentionJournalPath` provided by Tentacle. The journal will now be stored at the tentacle's root directory along with the existing `DeploymentJournal.xml`.

## How to test

1. Start Octopus server
2. Create a Tentacle using the latest `main` branch from `OctopusTentacle` (which has the new environment variable)
3. Add a package to the built-in feed (or use one from an external feed)
3. Add a "Transfer a package" step, select the package and specify a transfer destination
4. Set up project variables to debug Calamari (thanks Mark for the [guide here](https://trello.com/c/jS95BzZ3/10-how-do-i-debug-calamari))
4. Create a release and deploy it
5. Attach Calamari (make sure to build) to the process
6. `PackageRetentionJournal.json` should be created in the root directory of the tentacle

## Older Tentacle versions
~~If a Tentacle is created using a version prior to one that has the environment variable, then the deployment will blow up `System.ArgumentNullException: Value cannot be null. (Parameter 'path')`. How should we deal with versions that don't have the environment variable? Have a default value? Do we need to deal with older versions?~~ If you use an older Tentacle version it will default to using `TentacleHome` as the base path and `PackageRetentionJournal.json` as the name of the file (which is what the environment variable is set to in Tentacle anyway).
